test/e2e_node/swap_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	// Dir to enable swap.
+	swapDir          = "/mnt/swap"
+	totalSwapSize    = 512 // 512Mi
+	reservedSwapSize = "256Mi"
+)
```

Comment:
I would suggest that we use infrastructure with swap already enabled rather than turning it on here in the e2e, and skipping the test if that's not available. We will need the feature flag enabled as well.

Reply:
+1 - we should add a serial-swap lane, but might wanna tag these so we can have a reasonably focused run. Might be a good way to exercise the new file based Kubelet config? (for setting behavior and the like)

Reply:
Done.